tensorflow/python/layers/core.py

```diff
@@ -368,11 +375,19 @@ class Flatten(base.Layer):
   ```
   """
 
-  def __init__(self, **kwargs):
+  def __init__(self, data_format='channels_last', **kwargs):
     super(Flatten, self).__init__(**kwargs)
     self.input_spec = base.InputSpec(min_ndim=2)
+    self.data_format = utils.normalize_data_format(data_format)
 
   def call(self, inputs):
+    if self.data_format == 'channels_first':
+      # Ensure works for any dim, plus 2 due to [N, C, ...]
+      permutation = [0]
+      permutation.extend([i + 2 for i in
+                          range(len(inputs.shape.as_list()[2:]))])
```

Comment:
`inputs.shape.as_list()` looks incorrect. It should be `inputs.get_shape().as_list()`. Also take into account the case where `inputs.get_shape()` returns None (in this case, do nothing).

Reply:
Could catch, I should be using the getter.

I've added the check to ensure the shape is fully defined after the batch dim.
Would it be worth adding a `tf.logging.warning` mentioning the possible side-effects of running it in this way?